README.md

```diff
@@ -11,12 +11,12 @@ Prop supports two limiting strategies:
 To get going with Prop, you first define the read and write operations. These define how you write a registered request and how to read the number of requests for a given action. For example, do something like the below in a Rails initializer:
 
 ```ruby
-Prop.read do |key|
-  Rails.cache.read(key)
-end
-
-Prop.write do |key, value|
-  Rails.cache.write(key, value)
+# TODO Prop.cache = Rails.cache ... and deprecate old write / read
+Prop.read { |key| Rails.cache.read(key) }
+Prop.write { |key, value| Rails.cache.write(key, value) }
+Prop.increment do |key, value| 
+  # alternatively implement this with write(read + value), but you will suffer from race conditions
+  Rails.cache.increment(key, value) || Rails.cache.write(key, value)
```

Comment:
This would rely on users knowing what they do, which is hard to enforce.
I'm planing to change the interface to Prop.cache = Rails.cache and have
nice deprecation warnings when using .write / .read

On Tue, Oct 20, 2015 at 9:48 AM, Morten Primdahl <notifications@github.com>
wrote:

> In README.md <https://github.com/zendesk/prop/pull/7#discussion_r42521282>
> :
>
> > @@ -11,12 +11,12 @@ Prop supports two limiting strategies:
> >  To get going with Prop, you first define the read and write operations. These define how you write a registered request and how to read the number of requests for a given action. For example, do something like the below in a Rails initializer:
> >
> >  ```ruby
> > -Prop.read do |key|
> > -  Rails.cache.read(key)
> > -end
> > -
> > -Prop.write do |key, value|
> > -  Rails.cache.write(key, value)
> > +# TODO Prop.cache = Rails.cache ... and deprecate old write / read
> > +Prop.read { |key| Rails.cache.read(key) }
> > +Prop.write { |key, value| Rails.cache.write(key, value) }
> > +Prop.increment do |key, value|
> > +  # alternatively implement this with write(read + value), but you will suffer from race conditions
> > +  Rails.cache.increment(key, value) || Rails.cache.write(key, value)
>
> Why do you need to differentiate increment from write, changing the
> external interface shouldn't be required? Prop.write { |key, value|
> Rails.cache.increment(key, value) || Rails.cache.write(key, value) }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/prop/pull/7/files#r42521282>.
>